File: petrocity-api/src/main/java/br/com/magnasistemas/petrocityapi/corda/flow/PetrocityFlow.java
package br.com.magnasistemas.petrocityapi.corda.flow;

import co.paralleluniverse.fibers.Suspendable;
import net.corda.core.flows.*;
import net.corda.core.utilities.ProgressTracker;
import net.corda.core.contracts.Command;
import net.corda.core.identity.Party;
import net.corda.core.transactions.SignedTransaction;
import net.corda.core.transactions.TransactionBuilder;
import java.security.PublicKey;
import java.util.Arrays;
import java.util.List;

import br.com.magnasistemas.petrocityapi.corda.contract.*;
import br.com.magnasistemas.petrocityapi.corda.states.*;

public class PetrocityFlow extends FlowLogic<Void> {
	
	private final Party banco; 
	private final Party interessado;
	private final String cliente;
	private final int valor;
	private final String empreendimento;
	private final String localizacao;
	
	
	private final ProgressTracker progressTracker =  new ProgressTracker();
	
	public PetrocityFlow(Party banco, Party interessado, String cliente, int valor, String empreendimento, String localizacao) {
		this.banco = banco;
		this.interessado = interessado;
		this.cliente = cliente;
		this.valor = valor;
		this.empreendimento = empreendimento;
		this.localizacao = localizacao;
	
	}
	
	@Suspendable
	@Override
	public Void call() throws FlowException {
		// TODO Auto-generated method stub
		Party notary = getServiceHub().getNetworkMapCache().getNotaryIdentities().get(0);
		
		PetrocityState outputState = new PetrocityState(banco, interessado, cliente, valor, empreendimento, localizacao);
		//List<PublicKey> requiredSigners = Arrays.asList()
		// ###############################################################
		
		 List<PublicKey> requiredSigners = Arrays.asList(banco.getOwningKey(),interessado.getOwningKey());
	        Command command = new Command<>(new PetrocityContract.Create(), requiredSigners);

	        //crete transaction builder and add components
	        TransactionBuilder txBuilder = new TransactionBuilder(notary)
	                .addOutputState(outputState, PetrocityContract.ID)
	                .addCommand(command);

	        //verify the transaction
	        txBuilder.verify(getServiceHub());

	        //signing the transaction
	        SignedTransaction signedTx = getServiceHub().signInitialTransaction(txBuilder);

	        //create a session with other party
	        FlowSession otherPartySession = initiateFlow(banco);

	        SignedTransaction fullySignedTx = subFlow(new CollectSignaturesFlow(
	                signedTx, Arrays.asList(otherPartySession), CollectSignaturesFlow.tracker()
	        ));

	        //finalize transaction and send it to the counterparty
	        subFlow(new FinalityFlow(fullySignedTx, otherPartySession));

		
		
		// ###############################################################
		
		return null;
	}

	@Override
	public ProgressTracker getProgressTracker() {
		return progressTracker;
	}
	
	public static void main(String[] args) {
		
		System.out.println("Deu certo!!!!!!!!!!!!");
		
	}

}

File: cordapp-template-kotlin-petrocity/workflows/src/main/kotlin/com/template/flows/PetrocityFlow.kt
package com.template.flows

import co.paralleluniverse.fibers.Suspendable
import com.template.contracts.PetrocityContract
import com.template.states.PetrocityState
import net.corda.core.contracts.Command
import net.corda.core.contracts.UniqueIdentifier
import net.corda.core.contracts.requireThat
import net.corda.core.flows.*
import net.corda.core.identity.Party
import net.corda.core.node.ServiceHub
import net.corda.core.transactions.SignedTransaction
import net.corda.core.transactions.TransactionBuilder

@InitiatingFlow
@StartableByRPC
class PetrocityIssueInitiator(
        val banco: Party,
        val interessado: Party,
        val cliente: String,
        val valor: Int,
        val empreendimento: String,
        val local: String
) : FlowLogic<SignedTransaction>() {
    @Suspendable
    override fun call(): SignedTransaction {

        val notary = serviceHub.networkMapCache.notaryIdentities.first()
        val command = Command(PetrocityContract.Commands.Issue(), listOf(banco, interessado).map { it.owningKey })
        val PetrocityState = PetrocityState(
                banco,
                interessado,
                cliente,
                valor,
                empreendimento,
                local,
                UniqueIdentifier()
        )

        val txBuilder = TransactionBuilder(notary)
                .addOutputState(PetrocityState, PetrocityContract.ID)
                .addCommand(command)

        txBuilder.verify(serviceHub)
        val tx = serviceHub.signInitialTransaction(txBuilder)

        val sessions = (PetrocityState.participants - ourIdentity).map { initiateFlow(it as Party) }
        val stx = subFlow(CollectSignaturesFlow(tx, sessions))
        return subFlow(FinalityFlow(stx, sessions))
    }
}

@InitiatedBy(PetrocityIssueInitiator::class)
class PetrocityIssueResponder(val counterpartySession: FlowSession) : FlowLogic<SignedTransaction>() {

    @Suspendable
    override fun call(): SignedTransaction {
        val signedTransactionFlow = object : SignTransactionFlow(counterpartySession) {
            override fun checkTransaction(stx: SignedTransaction) = requireThat {
                val output = stx.tx.outputs.single().data
                "The output must be a PetrocityState" using (output is PetrocityState)
            }
        }
        val txWeJustSignedId = subFlow(signedTransactionFlow)
        return subFlow(ReceiveFinalityFlow(counterpartySession, txWeJustSignedId.id))
    }
}